pkg/cloud/services/gc/ec2.go

```diff
@@ -67,3 +70,33 @@ func (s *Service) deleteSecurityGroup(ctx context.Context, securityGroupID strin
 
 	return nil
 }
+
+// getProviderOwnedSecurityGroups gets cloud provider created security groups of ELBs for this cluster, filtering by tag: kubernetes.io/cluster/<cluster-name>:owned and VPC Id.
+func (s *Service) getProviderOwnedSecurityGroups(ctx context.Context) ([]*AWSResource, error) {
+	input := &ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			filter.EC2.ProviderOwned(s.scope.KubernetesClusterName()),
+		},
+	}
+
+	resources := []*AWSResource{}
+	err := s.ec2Client.DescribeSecurityGroupsPages(input, func(out *ec2.DescribeSecurityGroupsOutput, last bool) bool {
+		for _, group := range out.SecurityGroups {
+			if group != nil {
+				arn := composeFakeArn(sgService, sgResourcePrefix+*group.GroupId)
+				resource, err := composeAWSResource(aws.String(arn), converters.TagsToMap(group.Tags))
+				if err != nil {
+					s.scope.Error(err, "compose aws resource error: %v")
+					return false
+				}
+				resources = append(resources, resource)
+			}
```

Comment:
```suggestion
			if group == nil {
				continue
			}

			arn := composeFakeArn(sgService, sgResourcePrefix+*group.GroupId)
			resource, err := composeAWSResource(aws.String(arn), converters.TagsToMap(group.Tags))
			if err != nil {
				s.scope.Error(err, "compose aws resource error: %v")
				return false
			}
			resources = append(resources, resource)

```

Reply:
How about I remove `if group == nil`?
I come up that the describe result out.SecurityGroups should be have values or empty and the values (the group) should have no chance to be nil. 
I refer to https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/securitygroup/securitygroups.go#L330 before.

Reply:
Yep, agreed. I was thinking about why it was there. :D But I agree with you. Removing it should be okay. :)